content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +36,581 @@ are covered here as well as the `System.Diagnostics` API.
+var eventTags = new ActivityTagsCollection
 {
-    { "foo", 1 },
-    { "bar", "Hello, World!" },
-    { "baz", new int[] { 1, 2, 3 } }
+    { "operation", "calculate-pi" },
+    { "result", 3.14159 }
 };
 
-myActivity?.AddEvent(new("Gonna try it!", DateTimeOffset.Now, new(eventTags)));
+activity?.AddEvent(new("End Computation", DateTimeOffset.Now, eventTags));
 ```
 
-### Adding links
+### Create Activities with links
+
+A [Span](/docs/concepts/signals/traces/#spans) may be linked to zero or more
+other Spans that are causally related via a
+[Span Link](/docs/concepts/signals/traces/#span-links). Links can be used to
+represent batched operations where a Span was initiated by multiple initiating
+Spans, each representing a single incoming item being processed in the batch.
 
-An `Activity` can be created with zero or more
-[`ActivityLink`s](/docs/concepts/signals/traces/#span-links) that are causally
-related.
 
 ```csharp
-// Get a context from somewhere, perhaps it's passed in as a parameter
-var activityContext = Activity.Current!.Context;
 
 var links = new List<ActivityLink>
 {
-    new ActivityLink(activityContext)
+    new ActivityLink(activityContext1),
+    new ActivityLink(activityContext2),
+    new ActivityLink(activityContext3)
 };
 
-using var anotherActivity =
-    MyActivitySource.StartActivity(
-        ActivityKind.Internal,
-        name: "anotherActivity",
-        links: links);
-
-// do some work
+var activity = MyActivitySource.StartActivity(
+    ActivityKind.Internal,
+    name: "activity-with-links",
+    links: links);
 ```
 
+For more details how to read context from remote processes, see
+[Context Propagation](#context-propagation).
+
+
 ### Set Activity status
 
 {{% docs/languages/span-status-preamble %}}
 
-```csharp
-using var myActivity = MyActivitySource.StartActivity("SayHello");
+A [status](/docs/concepts/signals/traces/#span-status) can be set on a
+[span](/docs/concepts/signals/traces/#spans), typically used to specify that a
+span has not completed successfully - `SpanStatus.Error`. 
 
-try
-{
-	// do something
-}
-catch (Exception ex)
+By default, all spans are `Unset`, which means a span completed without error. The `Ok` status is reserved for when you need to explicitly mark a span as successful rather than stick with the default of Unset (i.e., “without error”).
+
+The status can be set at any time before the span is finished.
+
+It can be a good idea to record exceptions when they happen. It's recommended to
+do this in conjunction with
+[setting span status](/docs/specs/otel/trace/api/#set-status).
+
+```csharp
+private int rollOnce()
 {
-    myActivity.SetStatus(ActivityStatusCode.Error, "Something bad happened!");
+    int result;
+    var childActivity = MyActivitySource.StartActivity("rollOnce");
+
+    try
+    {
+        result = Random.Shared.Next(min, max + 1);
+    }
+    catch (Exception ex)
+    {
+        childActivity?.SetStatus(ActivityStatusCode.Error, ex.Message);
```

Comment:
is it recommended to set status description as ex.Message?

Reply:
ohh no. 
It is probably not even a good idea 🙂.